numpy/core/tests/test_numeric.py

```diff
@@ -2021,6 +2022,78 @@ def test_results(self):
             assert_(not res.flags['OWNDATA'])
 
 
+class TestMoveaxis(TestCase):
+    def test_move_to_end(self):
+        x = np.random.randn(5, 6, 7)
+        for source, expected in [(0, (6, 7, 5)),
+                                 (1, (5, 7, 6)),
+                                 (2, (5, 6, 7)),
+                                 (-1, (5, 6, 7))]:
+            actual = np.moveaxis(x, source, -1).shape
+            assert_(actual, expected)
```

Comment:
The tests could also check that the arrays share memory, but not a big deal, it is obviously enough they should.